pennylane/devices/qubit/adjoint_jacobian.py

```diff
@@ -33,14 +34,29 @@ def _dot_product_real(bra, ket, num_wires):
     return qml.math.real(qml.math.sum(qml.math.conj(bra) * ket, axis=sum_axes))
 
 
-def adjoint_jacobian(tape: QuantumTape):  # pylint: disable=too-many-statements
+def _get_output_ket(tape):
+    """Helper function to get the output state of a tape"""
+
+    # Initialization of state
+    prep_operation = None if len(tape._prep) == 0 else tape._prep[0]
+    ket = create_initial_state(
+        wires=tape.wires, prep_operation=prep_operation
+    )  #  ket(0) if prep_operation is None, else
+    for op in tape._ops:
```

Comment:
```suggestion
    for op in tape.operations[bool(prep_operation): ]:
```

starts at zero if no state preparation, and one if has a state preparation.